### PR TITLE
[WIP][SPARK-38213][CORE] Adding KafkaSink Metrics feature

### PR DESCRIPTION
**Pull request by senthh**
### What changes were proposed in this pull request?

KafkaSink Metrics feature is added
### Why are the changes needed?

KafkaSink Metrics feature will be useful to collect and store the Spark application's metrics into Kafka topic.
### Does this PR introduce _any_ user-facing change?

Yes
### How was this patch tested?

1. Created  test case
2. also tested manually in local cluster too

### CODE REVIEW

**Comment by senthh** [comment]
@dongjoon-hyun @HyukjinKwon @cloud-fan Could you pls review this MR?

File: dev/deps/spark-deps-hadoop-2-hive-2.3 [hunk 1]
```diff
@@ -161,6 +161,7 @@ jsp-api/2.1//jsp-api-2.1.jar
 jsr305/3.0.0//jsr305-3.0.0.jar
 jta/1.1//jta-1.1.jar
 jul-to-slf4j/1.7.32//jul-to-slf4j-1.7.32.jar
+kafka-clients/3.1.0//kafka-clients-3.1.0.jar
```
Comment (dongjoon-hyun) [inline dev/deps/spark-deps-hadoop-2-hive-2.3:164]: This is the evidence of the breaking dependency change.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]

File: core/pom.xml [hunk 2]
```diff
@@ -501,7 +501,11 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-crypto</artifactId>
     </dependency>
-
+    <dependency>
+      <groupId>org.apache.kafka</groupId>
+      <artifactId>kafka-clients</artifactId>
+      <version>${kafka.version}</version>
+    </dependency>
```
[the remark this answers is not in the document]
Comment (senthh) [reply core/pom.xml:508]: @dongjoon-hyun Thanks for pointing this.  Moving this KafkaSink metrics features into external module will be correct?

**Comment by senthh** [comment]
Creating new PR so I m closing this PR

**Pull request closed without merging**